centreon/www/front_src/src/Resources/Graph/Performance/TimePeriods/index.cypress.spec.tsx

```diff
@@ -0,0 +1,471 @@
+      lastDay: { day: days[0], indexDayInRowWeek: 0, value: 30 },
+      numberWeeks: 6
+    }
+  },
+  {
+    March: {
+      date: '2023-03-01T12:59:41.041Z',
+      firstDay: { day: days[3], indexDayInRowWeek: 3, value: 1 },
+      lastDay: { day: days[5], indexDayInRowWeek: 5, value: 31 },
+      numberWeeks: 5
+    }
+  },
+  {
+    February: {
+      date: '2023-02-01T12:59:41.041Z',
+      firstDay: { day: days[3], indexDayInRowWeek: 3, value: 1 },
+      lastDay: { day: days[2], indexDayInRowWeek: 2, value: 28 },
+      numberWeeks: 5
+    }
+  },
+  {
+    January: {
+      date: '2023-01-01T12:59:41.041Z',
+      firstDay: { day: days[0], indexDayInRowWeek: 0, value: 1 },
+      lastDay: { day: days[2], indexDayInRowWeek: 2, value: 31 },
+      numberWeeks: 5
+    }
+  }
+];
+
+const dataTest = [
+  {
+    button: 'Next month',
+    data: months2023,
+    initialDate: '2023-01-01T12:59:41.041Z',
+    timezone: 'Europe/Paris'
+  },
+  {
+    button: 'Previous month',
+    data: month2023Reverse,
+    initialDate: '2023-12-01T12:59:41.041Z',
+    timezone: 'Europe/Paris'
+  }
+];
+
+enum ButtonCalendar {
+  NMONTH = 'Next month',
+  PMONTH = 'Previous month'
+}
+interface GetPreviousNextMonth {
+  currentMonth: string;
+  labelButton: string;
+}
+
+const getPreviousNextMonth = ({
+  currentMonth,
+  labelButton
+}: GetPreviousNextMonth): void => {
+  if (equals(labelButton, ButtonCalendar.PMONTH)) {
+    if (!equals(currentMonth, 'January')) {
+      cy.get(`[aria-label="${labelButton}"]`).click();
+    }
+  }
+  if (equals(labelButton, ButtonCalendar.NMONTH)) {
+    if (!equals(currentMonth, 'December')) {
+      cy.get(`[aria-label="${labelButton}"]`).click();
+    }
+  }
+};
+
+const changeDate = (): void => undefined;
+
+const setStart = undefined;
+const checkIfDuplicateExists = (arr: Array<unknown>): boolean => {
+  return new Set(arr).size !== arr.length;
+};
+
+dataTest.forEach((item) =>
+  describe(`calendar`, () => {
+    before(() => {
+      const userData = renderHook(() => useAtomValue(userAtom));
+
+      act(() => {
+        userData.result.current.timezone = item.timezone;
+        userData.result.current.locale = 'en_US';
+      });
+    });
+
+    beforeEach(() => {
+      const { result } = renderHook(() => useDateTimePickerAdapter());
+
+      act(() => {
+        const { Adapter } = result.current;
+        cy.mount({
+          Component: (
+            <LocalizationProvider adapterLocale="en" dateAdapter={Adapter}>
+              <DateTimePickerInput
+                changeDate={changeDate}
+                date={new Date(item.initialDate)}
+                property={CustomTimePeriodProperty.start}
+                setDate={setStart}
+              />
+            </LocalizationProvider>
+          )
+        });
+      });
+    });
+
+    it('checks input calendar value contains correct date', () => {
+      const { result } = renderHook(() => useLocaleDateTimeFormat());
+      act(() => {
```

Comment:
Is it mandatory to wrap it inside an act?
act is meant to be used when you update a state or something